pydantic_ai_slim/pydantic_ai/output.py

```diff
@@ -0,0 +1,289 @@
+from __future__ import annotations
+
+from collections.abc import Awaitable, Sequence
+from dataclasses import dataclass
+from typing import Any, Callable, Generic, Literal, Union
+
+from typing_extensions import TypeAliasType, TypeVar, get_args, get_origin
+from typing_inspection import typing_objects
+from typing_inspection.introspection import is_union_origin
+
+from .messages import RetryPromptPart
+from .tools import RunContext
+
+OutputDataT = TypeVar('OutputDataT', default=str, covariant=True)
+"""Covariant type variable for the result data type of a run."""
+
+T = TypeVar('T')
+
+OutputMode = Literal['text', 'tool', 'model_structured', 'prompted_structured', 'tool_or_text']
+"""All output modes."""
+StructuredOutputMode = Literal['tool', 'model_structured', 'prompted_structured']
+"""Output modes that can be used for structured output. Used by ModelProfile.default_structured_output_mode"""
+
+
+class ToolRetryError(Exception):
```

Comment:
if this is internal, it should probably be in `_output.py`.

Reply:
It's used in a docs example, so I'll leave it here